src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Analyzers/ComponentParameterNullableWarningSuppressor.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Resources;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.Razor.Compiler.Analyzers;
+
+#pragma warning disable RS1041 // Compiler extensions should be implemented in assemblies targeting netstandard2.0
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public class ComponentParameterNullableWarningSuppressor : DiagnosticSuppressor
+{
+    private static readonly LocalizableString Description = new LocalizableResourceString(nameof(AnalyzerResources.ComponentParameterNullableWarningSuppressorDescription), AnalyzerResources.ResourceManager, typeof(AnalyzerResources));
+
+    public override ImmutableArray<SuppressionDescriptor> SupportedSuppressions => [
+            new SuppressionDescriptor("RZS1001", "CS8618", Description)
```

Comment:
Do we expect to have like a separate Suppressor class per each diagnostic we want to suppress? If yes, perhaps we should have some static class grouping all the diagnostic numbers like RZS1001 in one place (so we know which one comes next). If not, perhaps ComponentParameterNullableWarningSuppressor should be renamed to be more generic. Although either of these can be done if/when we add the next suppressor, of course.

Reply:
Yeah, I don't know what else we'll have. I guess it doesn't hurt to pull it out into a static class now though just to keep things neat.